pkg/controller/controller_ref_manager.go

```diff
@@ -142,3 +143,103 @@ func (m *PodControllerRefManager) ReleasePod(pod *api.Pod) error {
 	}
 	return err
 }
+
+type ReplicaSetControllerRefManager struct {
+	rsControl          RSControlInterface
+	controllerObject   api.ObjectMeta
+	controllerSelector labels.Selector
+	controllerKind     unversioned.GroupVersionKind
+}
+
+// NewReplicaSetControllerRefManager returns a ReplicaSetControllerRefManager that exposes
+// methods to manage the controllerRef of ReplicaSets.
+func NewReplicaSetControllerRefManager(
+	rsControl RSControlInterface,
+	controllerObject api.ObjectMeta,
+	controllerSelector labels.Selector,
+	controllerKind unversioned.GroupVersionKind,
+) *ReplicaSetControllerRefManager {
+	return &ReplicaSetControllerRefManager{rsControl, controllerObject, controllerSelector, controllerKind}
+}
+
+// Classify first filters out inactive ReplicaSets, then it classify the remaining ReplicaSets
```

Comment:
Inactive pods will be garbage-collected by the podGC controller. We don't have anything similar for replica sets so I don't think we should filter out any replica sets at this point - @caesarxuchao to keep me correct